Assets/Scripts/DragDropToDistance.cs

```diff
@@ -0,0 +1,268 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.InputSystem;
+
+public class DragDropToDistance : MonoBehaviour
+{
+    //followed this tutorial for basic touchscreen dragging: https://www.youtube.com/watch?v=zo1dkYfIJVg
+
+    //variables
+    [SerializeField] private InputAction press;
+    [SerializeField] private InputAction screenPosition;
+    [SerializeField] private Camera playerCamera;
+    [SerializeField] private GameObject yellowHighlight;
+    [SerializeField] private GameObject redHighlight;
+    [SerializeField] private GameObject greenCorrectIcon;
+    private ShapeSorting_GameManager gameManagerScript;
+    private Vector3 centrePosition = new Vector3(-0.614f, 11.33f, 7.18f);
+    private bool noTriggers;
+
+    private Vector3 currentScreenPosition;
+    private float targetZ;
+    private float cameraDifferential = 5.5f;
+    private Rigidbody objectRB;
+    private bool isDragging;
+
+    private float moveSpeed = 0.9f;
+    [SerializeField] private string targetTag;
+
+    private bool isMoving = false; // Flag to track whether movement is in progress
+    private bool isAtMiddle = false;//to check whether object has reached middle;
```

Comment:
Could we change this to 'isShapeAtMiddleOfShapeSpace' or something indicating where the middle is?